packages/astro/src/compiler/codegen/index.ts

```diff
@@ -47,6 +47,22 @@ interface CodeGenOptions {
   fileID: string;
 }
 
+interface HydrationAttributes {
+  method?: 'load' | 'idle' | 'visible';
```

Comment:
This should be required and not optional. We intentionally want to make client loading an *intentional* choice and we also don't want any of these to be the default (to force devs to think about which one is appropriate for the use-case).

Reply:
Or am I misunderstanding this? If there is no method does it SSR only?

Reply:
Yeah the default undefined state just means no hydration.  Making `method` required would mean findHydrationAttributes` would have to return `undefined` and tweaking the `getComponentWrapper` parameters to handle `HydrationAttributes | undefined`

Reply:
Ok cool, disregard this comment then.